src/io/pipe.rs

```diff
@@ -53,3 +60,64 @@ pub fn pipe() -> io::Result<(OwnedFd, OwnedFd)> {
 pub fn pipe_with(flags: PipeFlags) -> io::Result<(OwnedFd, OwnedFd)> {
     backend::io::syscalls::pipe_with(flags)
 }
+
+/// `splice(fd_in, off_in, fd_out, off_out, len, flags)`—Transfer data between a file and a pipe.
+///
+/// This function transfers up to `len` bytes of data from the file descriptor `fd_in`
+/// to the file descriptor `fd_out`, where one of the file descriptors
+/// must refer to a pipe.
+///
+/// `off_*` must be `None` if the corresponding fd refers to a pipe.
+/// Otherwise its value is the starting offset to the file,
+/// from which the data is read/written.
+/// passing `None` causes the read/write to start from the file offset,
+/// and the file offset is adjusted appropriately.
+///
+/// # References
+///  - [Linux]
+///
+/// [Linux]: https://man7.org/linux/man-pages/man2/splice.2.html
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[inline]
+pub fn splice<FdIn: AsFd, FdOut: AsFd>(
+    fd_in: FdIn,
+    off_in: Option<u64>,
+    fd_out: FdOut,
+    off_out: Option<u64>,
```

Comment:
In the linux syscall, `off_in` and `off_out` are in-out parameters, that return how many bytes were read and written, respectively.

Reply:
where is that written? according to the [manual](https://man7.org/linux/man-pages/man2/splice.2.html) the number of bytes read/written is the return value on success, with no mention of `off_*` being used for output.

Reply:
Empirically,`*off_in` is indeed written to by the system call. I don't actually know if the change in the value can ever differ from the returned size though, so perhaps it's redundant? I guess if it is indeed redundant, then we don't need to expose it here, though in that case, it's worth documenting that that aspect of the Linux syscall isn't reflected in the rustix API.

Reply:
i would argue that if this property itself is not documented, then its not worth documenting the changes to it.
it would just be confusing.